framework/src/main/java/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -237,7 +244,14 @@ private void handleNullTypeArguments(
                 if (withoutNullResult == null) {
                     // withoutNullResult is null when the only constraint on a type argument is
                     // where a method argument is null.
-                    withoutNullResult = typeFactory.getUninferredWildcardType(atv);
+                    if (objectWithAnnosFromNull == null) {
+                        Elements elements = typeFactory.getProcessingEnv().getElementUtils();
+                        TypeMirror objectTM = elements.getTypeElement("java.lang.Object").asType();
+                        objectWithAnnosFromNull =
+                                AnnotatedTypeMirror.createType(objectTM, typeFactory, false);
+                        objectWithAnnosFromNull.addAnnotations(result.getAnnotations());
```

Comment:
I'm confused why it is the right thing to use the annotations only from the first `result`. The loop goes through all type parameters, multiple of which might come from `null`, but only the first determines the value of `objectWithAnnosFromNull`.
Can you add an explanation?
Or should you ask the type factory what the type of `null` is, instead?

Reply:
Also, should `objectWithAnnosFromNull` be set to `null` again at some point?